apps/emqx/src/emqx_maybe.erl

```diff
@@ -0,0 +1,90 @@
+%%--------------------------------------------------------------------
+%% Copyright (c) 2017-2023 EMQ Technologies Co., Ltd. All Rights Reserved.
+%%
+%% Licensed under the Apache License, Version 2.0 (the "License");
+%% you may not use this file except in compliance with the License.
+%% You may obtain a copy of the License at
+%%
+%%     http://www.apache.org/licenses/LICENSE-2.0
+%%
+%% Unless required by applicable law or agreed to in writing, software
+%% distributed under the License is distributed on an "AS IS" BASIS,
+%% WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+%% See the License for the specific language governing permissions and
+%% limitations under the License.
+%%--------------------------------------------------------------------
+
+-module(emqx_maybe).
```

Comment:
I am not sure how I feel about this module... It has a distinct Haskell style to it, but on the other hand it doesn't conform with "maybe" semantics found in other functional programming languages, for example because it gets behavior of `emqx_maybe:t(undefined)` wrong.

I checked most cases where it's used, and usually it could be replaced with simpler, explicit code without adding too much boilerplate.